docs/Joins.md

```diff
@@ -122,4 +122,18 @@ is treated as a single stream and joined against `stream2` from `bolt3`.
 2. Duplication of joined records across windows is possible when using Sliding Windows. This is because the tuples continue to exist
    across multiple windows when using Sliding Windows.
 
-
+3. If message timeouts are enabled, ensure the timeout setting (topology.message.timeout.secs) is large enough to comfortably
+   accommodate the window size, plus the additional processing by other spouts and bolts.
+
+4. Joining a window of two streams with M and N elements each, *in the worst case*, can produce MxN elements with every output tuple
+   anchored to one tuples from each input stream. This can mean a lot of output tuples from JoinBolt and even more ACKs for downstream bolts
```

Comment:
Should not this be
produce MxN elements with every output tuple anchored to <s>one</s> tuples  from each input stream.